arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedColumnIterator.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.arrow.vectorized.parquet;
+
+import java.io.IOException;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.iceberg.arrow.vectorized.NullabilityHolder;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.Dictionary;
+import org.apache.parquet.column.page.DataPage;
+import org.apache.parquet.column.page.DictionaryPage;
+import org.apache.parquet.column.page.PageReadStore;
+import org.apache.parquet.column.page.PageReader;
+import org.apache.parquet.io.ParquetDecodingException;
+
+/**
+ * Vectorized version of the ColumnIterator that reads column values in data pages of a column in a row group in a
+ * batched fashion.
+ */
+public class VectorizedColumnIterator {
+
+  private final ColumnDescriptor desc;
+  private final VectorizedPageIterator vectorizedPageIterator;
+
+  // state reset for each row group
+  private PageReader columnPageReader = null;
+  private long totalValuesCount = 0L;
+  private long valuesRead = 0L;
+  private long advanceNextPageCount = 0L;
+  private final int batchSize;
+
+  public VectorizedColumnIterator(ColumnDescriptor desc, String writerVersion, int batchSize) {
+    this.desc = desc;
+    this.batchSize = batchSize;
+    this.vectorizedPageIterator = new VectorizedPageIterator(desc, writerVersion, batchSize);
+  }
+
+  public Dictionary setRowGroupInfo(PageReadStore store, boolean allPagesDictEncoded) {
+    this.columnPageReader = store.getPageReader(desc);
+    this.totalValuesCount = columnPageReader.getTotalValueCount();
+    this.valuesRead = 0L;
+    this.advanceNextPageCount = 0L;
+    this.vectorizedPageIterator.reset();
+    Dictionary dict = readDictionaryForColumn(store);
```

Comment:
This `readDictionaryForColumn` uses `desc` and calls `getPageReader` that is called just above to initialize `columnPageReader`. I think it would be better to use the same `readDictionary` static method that is used by the non-vectorized path. For one thing, it would eliminate the duplicate call to `getPageReader` because you'd pass `columnPageReader` in, and it would also get rid of the unknown use of `desc` because it is static. And we'd not need to maintain two copies of the method that do pretty much the same thing.